test/core/client_channel/lb_policy/xds_override_host_test.cc

```diff
@@ -74,6 +80,76 @@ TEST_F(XdsOverrideHostTest, NoConfigReportsError) {
       absl::InvalidArgumentError("Missing policy config"));
 }
 
+TEST_F(XdsOverrideHostTest, OverrideHost) {
+  const std::array<absl::string_view, 2> kAddresses = {"ipv4:127.0.0.1:441",
+                                                       "ipv4:127.0.0.1:442"};
+  EXPECT_EQ(policy_->name(), "xds_override_host_experimental");
+  // 1. We use pick_first as a child
+  EXPECT_EQ(ApplyUpdate(BuildUpdate(kAddresses,
+                                    MakeXdsOverrideHostConfig("round_robin")),
+                        policy_.get()),
+            absl::OkStatus());
+  ExpectConnectingUpdate();
+  EXPECT_NE(ExpectState(GRPC_CHANNEL_CONNECTING), nullptr);
```

Comment:
Reminders:
- Do not explicitly check that we get a certain number of repeat updates for the same connectivity state, because that's ultimately an implementation detail.  After `ExpectConnectingUpdate()`, we should never call `ExpectState(GRPC_CHANNEL_CONNECTING)`, unless there was some other state in between.  In general, the thing after `ExpectConnectingUpdate()` should be either `WaitForConnected()` or `WaitForConnectionFailed()`.
- Whenever we get a connectivity state update, we need to check the picker's behavior instead of just checking that the picker is non-null.  This is ultimately the thing that actually defines how the LB policy works, so if we're not checking it, then we have no guarantee that the LB policy is actually doing the right thing.

Please fix throughout.

Reply:
I think it's important to keep in mind that the whole goal of the LB policy is to return the right picks.  All of the subchannel connectivity management that we're testing here **exists only for the purpose of ensuring that the policy returns the right picks**.  Given that, I think it's very important to always verify exact picker behavior, because that's ultimately the thing we actually care about.  If that's wrong, then the test is useless, because it's not verifying that the LB policy is behaving correctly.

As long as we're relying on the picking behavior of simple underlying policies like pick_first and round_robin, I don't think it's going to add any particular maintenance burden to the tests to check the pick results, because the picking behavior of those two policies is not going to change.  And if we're already going to assert that the picker returns complete picks, I don't think we gain anything by not actually checking that the resulting address is the one we expect.

Please change all occurrances of `EXPECT_TRUE(ExpectPickComplete(picker.get()).has_value())` to instead do `EXPECT_EQ(ExpectPickComplete(picker.get()), "ipv4:127.0.0.1:441")` (or whatever address is appropriate).